src/common.jl

```diff
@@ -0,0 +1,11 @@
+Base.isfinite(a::AbstractVec) = all(isfinite.(a))
+Base.isinf(a::AbstractVec) = any(isinf.(a))
+
+function StaticArrays.similar_type(::Type{V}, ::Type{Float64}, size::Size) where {V<:AbstractVec}
```

Comment:
Why not:
```julia
function StaticArrays.similar_type(::Type{V}, ::Type{F}, size::Size) where {V<:AbstractVec, F<:AbstractFloat}
    if size == Size(V) && eltype(V) == F
        return V
    end
end
```
That should atleast work with all kinds of floats?

Reply:
good call - implemented this. There were some weird things going on when I tried to do it for all `Real`. Will fix that eventually.